.travis.yml

```diff
@@ -1,10 +1,35 @@
 language: java
-jdk:
-  - openjdk8
-  - oraclejdk8
-dist: trusty
 
-install: true
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
 
-script:
- - ./gradlew build
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
+jobs:
+ include:
+    - stage: testandvalidate
+      jdk: openjdk8
+      script: ./gradlew check
+    - jdk: oraclejdk8
+      script: ./gradlew check
+    - jdk: oraclejdk8
+      env: validate
```

Comment:
if stage is named 'validate' do we need an env: validate anymore?

Reply:
just a marker, as it is the only one of the three parallel tasks that runs code analysis
can be removed if not needed